runtime/compiler/trj9/x/env/J9CPU.hpp

```diff
@@ -62,34 +63,23 @@ class CPU : public J9::CPU
 protected:
 
    CPU() :
-         J9::CPU(),
-         _featureFlags(0),
-         _featureFlags2(0),
-         _featureFlags8(0)
-      {}
+         J9::CPU()
+      {
+      }
 
 public:
 
-   const char *getX86ProcessorVendorId(TR::Compilation *comp);
-   uint32_t getX86ProcessorSignature(TR::Compilation *comp);
-   uint32_t getX86ProcessorFeatureFlags(TR::Compilation *comp);
-   uint32_t getX86ProcessorFeatureFlags2(TR::Compilation *comp);
-   uint32_t getX86ProcessorFeatureFlags8(TR::Compilation *comp);
+   bool testOSForSSESupport();
+   bool getX86SupportsSSE4_1();
+   bool getX86SupportsHLE();
+   bool getX86SupportsPOPCNT();
+
```

Comment:
getX86SupportsSSE4_1(), getX86SupportsHLE(), getX86SupportsPOPCNT() are latency functions. They are duplicate to these feature checks. https://github.com/eclipse/omr/blob/b3e2f861e48894b42751e943e197606d685d50a4/compiler/x/codegen/OMRCodeGenerator.hpp#L169
It may be a good time to get ride of these completely.

Reply:
Hi Victor, thanks for the review. For getX86SupportsHLE(), it is easy to remove and replace because the HLE feature checks only happen in x86 specific files. But for getX86SupportsSS4_1() and getX86SupportsPOPCNT(), the checks show up in cross-platform files, so if we don't declare them for all platforms or put Macro guards around them, we will get compile errors on non X86 platforms. What do you think we should do with those two?

Reply:
i.e. the feature check of getX86SupportsSSE4_1() shows up in env/VMJ9.cpp `if ((TR::Compiler->target.cpu.isX86() && comp->cg()->getX86ProcessorInfo().supportsSSE4_1())`